genie-agent/src/main/java/com/netflix/genie/agent/execution/process/impl/JobProcessManagerImpl.java

```diff
@@ -125,12 +127,15 @@ public void launchProcess(
         });
 
         // Validate arguments
-        if (commandLine == null) {
+        if (commandArguments == null) {
             throw new JobLaunchException("Job command-line arguments is null");
-        } else if (commandLine.isEmpty()) {
+        } else if (commandArguments.isEmpty()) {
```

Comment:
Can't you just combine these two into an `||` and say "Job command executable is missing".

I mean technically all these checks are redundant with our validation of the DTO's and whatnot upstream but I can understand being safe and making this unit self contained.

With `@ParametersAreNonNullByDefault` all these fields should be non-null warnings in the compiler but obviously that doesn't enforce runtime so if you want to be really secure in it I guess this works.

Reply:
I agree with this but i was trying not to change the logic beyond the strictly necessary.